### PR TITLE
can choose maven sub module in pushing image

### PR DESCRIPTION
**Pull request by Eskibear**

### CODE REVIEW

**Comment by Eskibear** [comment]
depend on #1170

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/PushImageAction.java [hunk 1]
```diff
@@ -50,22 +52,25 @@
...
     public void onActionPerformed(AnActionEvent event) {
-        Project project = event.getProject();
-        if (project == null) {
+        Module module = DataKeys.MODULE.getData(event.getDataContext());
+        if (module == null) {
             notifyError(Constant.ERROR_NO_SELECTED_PROJECT);
             return;
         }
-        ApplicationManager.getApplication().invokeLater(() -> runConfiguration(project));
+        Project project = module.getProject();
```
Comment (jdneo) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/PushImageAction.java]: project not used?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/pushimage/ui/SettingPanel.java [hunk 2]
```diff
@@ -179,6 +192,20 @@ private void setupArtifactCombo(List<Artifact> artifacts, String targetPath) {
         isCbArtifactInited = true;
     }
 
+    private void setupMavenProjectCombo(List<MavenProject> mvnprjs, String targetPath) {
```
Comment (jdneo) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/pushimage/ui/SettingPanel.java]: Can we also add module icon in the combo box item

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/PushImageAction.java [hunk 3]
```diff
@@ -50,22 +52,25 @@
...
 
     @SuppressWarnings({"deprecation", "Duplicates"})
-    private void runConfiguration(Project project) {
+    private void runConfiguration(Module module) {
+        Project project = module.getProject();
         final RunManagerEx manager = RunManagerEx.getInstanceEx(project);
         final ConfigurationFactory factory = configType.getPushImageRunConfigurationFactory();
         RunnerAndConfigurationSettings settings = manager.findConfigurationByName(
-                String.format("%s: %s", factory.getName(), project.getName()));
+                String.format("%s: %s:%s", factory.getName(), project.getName(), module.getName()));
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/PushImageAction.java]: need space before the last %s?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/PushImageAction.java [hunk 4]
```diff
@@ -50,22 +52,25 @@
...
         if (settings == null) {
-            settings = manager.createConfiguration(String.format("%s: %s", factory.getName(), project.getName()),
+            settings = manager.createConfiguration(
+                    String.format("%s: %s:%s", factory.getName(), project.getName(), module.getName()),
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/PushImageAction.java]: same as above

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/pushimage/ui/SettingPanel.java [hunk 5]
```diff
@@ -188,12 +215,18 @@ public void reset(PushImageRunConfiguration conf) {
...
+            setupMavenProjectCombo(mavenProjects, conf.getTargetPath());
         }
 
         PrivateRegistryImageSetting acrInfo = conf.getPrivateRegistryImageSetting();
         containerSettingPanel.setTxtFields(acrInfo);
         if (Utils.isEmptyString(conf.getDockerFilePath())) {
-            containerSettingPanel.setDockerPath(DockerUtil.getDefaultDockerFilePathIfExist(project));
+            MavenProject prj = (MavenProject) cbMavenProject.getSelectedItem();
+            if (prj != null) {
+                containerSettingPanel.setDockerPath(DockerUtil.getDefaultDockerFilePathIfExist(prj.getDirectory()));
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/pushimage/ui/SettingPanel.java]: just curious whether this part of code is already triggered by the select listener

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/utils/DockerUtil.java [hunk 6]
```diff
@@ -187,4 +187,14 @@ public static String getDefaultDockerFilePathIfExist(Project project) {
         }
         return "";
     }
+
+    public static String getDefaultDockerFilePathIfExist(String basePath) {
+        if (!Utils.isEmptyString(basePath)) {
+            Path targetDockerfile = Paths.get(basePath, Constant.DOCKERFILE_NAME);
+            if (targetDockerfile != null && targetDockerfile.toFile().exists()) {
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/utils/DockerUtil.java]: toFile will throw exception. shall we handle it here?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/PushImageAction.java (hunk 3 above)
Comment (Eskibear) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/PushImageAction.java]: I removed the whitespace intentionally, to combine projectName & moduleName as a group. Do you think we should reserve the space ?

Comment (adashen) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/PushImageAction.java]: just make sure it is as designed. No need to reserve the space

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/pushimage/ui/SettingPanel.java (hunk 2 above)
Comment (Eskibear) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/pushimage/ui/SettingPanel.java]: added

**Pull request merged**